src/Plugin/QueueWorker/HubspotCommerceQueue.php

```diff
@@ -0,0 +1,81 @@
+<?php
+
+namespace Drupal\hubspot_commerce\Plugin\QueueWorker;
+
+use Drupal\hubspot_commerce\Services\HubspotCommerce;
+
+use Drupal\Core\Annotation\QueueWorker;
+use Drupal\Core\Logger\LoggerChannelFactoryInterface;
+use Drupal\Core\Plugin\ContainerFactoryPluginInterface;
+
+use Symfony\Component\DependencyInjection\ContainerInterface;
+
+/**
+ * Class HubspotCommerceQueue.
+ *
+ * @QueueWorker(
+ *   id = "hubspot_commerce_queue",
+ *   title = @Translation("Hubspot Commerce queue"),
+ *   cron = {"time" = 60}
+ * )
+ */
+class HubspotCommerceQueue extends QueueWorker implements ContainerFactoryPluginInterface {
+
+  /**
+   * The Hubspot Commerce service.
+   *
+   * @var \Drupal\hubspot_commerce\Services\HubspotCommerce
+   */
+  protected $service;
+
+  /**
+   * The logger.
+   *
+   * @var \Psr\Log\LoggerInterface
+   */
+  protected $logger;
+
+  /**
+   * {@inheritdoc}
+   */
+  public function __construct(
+    array $configuration,
+    $plugin_id,
+    $plugin_definition,
+    HubspotCommerce $service,
+    LoggerChannelFactoryInterface $logger_factory
+  ) {
+    parent::__construct($configuration, $plugin_id, $plugin_definition);
+
+    $this->service = $service;
+    $this->logger = $logger_factory->get(HUBSPOT_COMMERCE_LOGGER_CHANNEL);
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public static function create(ContainerInterface $container, array $configuration, $plugin_id, $plugin_definition) {
+    return new static(
+      $configuration,
+      $plugin_id,
+      $plugin_definition,
+      $container->get('hubspot_commerce.service'),
+      $container->get('logger.factory')
+    );
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public function processItem($entity) {
+    try {
+      $this->service->syncQueueItem($entity);
```

Comment:
**CHANGE REQUEST**

The function provided by the entity does not have anything specific to the queue, it generally syncs an entity and it could be called elsewhere in the code including from other modules. Let's name the function just `->sync($entity);`